packages/alfa-css/src/grammars/color.ts

```diff
@@ -99,13 +99,49 @@ function rgbaColor(stream: Stream<Token>): Color {
   return color;
 }
 
+function hexColor(stream: Stream<Token>): Color {
+  const color: Mutable<Color> = { red: 0, green: 0, blue: 0, alpha: 1 };
+
+  const args = functionArguments(stream);
+
+  if (args.length !== 1 || args[0].type !== TokenType.Ident) {
+    return Transparent;
+  }
+
+  let hex = (<Ident>args[0]).value;
+
+  if (hex.length === 8) {
+    color.alpha = parseInt(hex.slice(6, 8), 16) / 255;
+    hex = hex.substring(0, 6);
+  }
+
+  if (hex.length === 4) {
+    color.alpha = parseInt(hex.slice(3, 4).repeat(2), 16) / 255;
+    hex = hex.substring(0, 3);
+  }
+
+  if (hex.length === 3) {
+    hex = hex[0] + hex[0] + hex[1] + hex[1] + hex[2] + hex[2];
+  }
+
+  if (hex.length !== 6) {
+    return Transparent;
+  }
+
+  const num = parseInt(hex, 16);
+  color.red = num >> 16;
+  color.green = (num >> 8) & 255;
+  color.blue = num & 255;
+
+  return color;
```

Comment:
The parsing will need to be more robust as it currently doesn't handle things like non-hex characters and overflows. Here's a sketch of a more robust, and slightly more performant, approach:

1. Figure out the "repetition count" of the given hex value based on its length. For short hand colors (e.g. `#123`) the repetition count is 2 whereas for long hand colors (e.g. `#123456`) the repetition count is 1). If a repetition count cannot be determined, i.e. the length of the value is incorrect, return `Transparent`.
2. Figure out if an alpha channel is specified (e.g. `#1234` or `#123456ab`).
3. Run through every character and convert it to a number. An example of this is already implemented here: https://github.com/Siteimprove/alfa/blob/master/packages/alfa-css/src/alphabet.ts#L294-L300. If a non-hex character is encountered, return `Transparent`.
4. Append the number `n`-times to the channel (red, green, blue, alpha) corresponding to the character, where `n` is the repetition count. We're dealing with hex values so the code would be along the lines of:

    ```ts
    channel = (channel * 16) + value;
    ```